src/ecmech/ECMech_evptn.h

```diff
@@ -81,15 +80,18 @@ namespace ecmech {
             __ecmech_host__
             inline void getParams(std::vector<double> & params
                                   ) const {
+#ifdef ECMECH_DEBUG
                // do not clear params in case adding to an existing set
                int paramsStart = params.size();
+#endif
 
                params.push_back(_c11);
                params.push_back(_c12);
                params.push_back(_c44);
 
-               int iParam = params.size() - paramsStart;
-               assert(iParam == nParams);
+#ifdef ECMECH_DEBUG
```

Comment:
Here and elsewhere, do we really need to put these assert checks inside of ECMECH_DEBUG ifdefs? These are not performance-critical parts of the code, and I would be tempted to have the asserts active even in non-debug builds unless they are causing some kind of trouble.

Reply:
So, the asserts are only active/compiled in debug builds. In release builds, the assert isn't compiled and so the check is never done. It's the reason why I hid this behind the ECMECH_DEBUG define so the compiler wouldn't complain about some variables never being used. If we want this assert active for all builds then we'll want to look at another function to do this.

Reply:
Got it. I educated myself a bit more on assert, and see what you mean. We should just make sure we test with debug builds often enough.